python/bolt-mcp/tests/test_streaming.py

```
@@ -0,0 +1,114 @@
+def _parse_all_sse(body: bytes) -> list[dict]:
+    """Return every JSON-RPC message carried by a (finite) SSE response body."""
+    text = body.decode("utf-8").replace("\r\n", "\n")
+    out: list[dict] = []
+    for block in text.split("\n\n"):
+        data = [line[len("data:") :].lstrip() for line in block.split("\n") if line.startswith("data:")]
+        if data:
+            out.append(msgspec.json.decode("\n".join(data).encode()))
+    return out
+
+
+def _call_crunch(client, session_id, *, json_response=False):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_call_crunch`

(ANN202)